File: atcoder/abc211C_chokudai.py
# uninhm
# https://atcoder.jp/contests/abc211/tasks/abc211_c
# dp

word = "chokudai"
s = input()

dp = [0] + [1]*len(s)
mod = 10**9+7
for l in word:
    for i, e in enumerate(s, start=1):
        if e == l:
            dp[i] = (dp[i-1] + dp[i]) % mod
        else:
            dp[i] = dp[i-1]

print(dp[-1])

File: atcoder/abc211C_chokudai.nim
# Vicfred
# https://atcoder.jp/contests/abc211/tasks/abc211_c
# dynamic programming

import sequtils

const modulo = 1000000007

let S = stdin.readLine

let n = S.len

var dp = newSeqWith(n + 1, newSeq[int](8 + 1))

for i in 0..n:
  dp[i][0] = 1

const C = "chokudai"

for i in 1..n:
  for j in 1..8:
    if S[i - 1] != C[j - 1]:
      dp[i][j] = dp[i - 1][j]
    else:
      dp[i][j] = dp[i - 1][j - 1] + dp[i - 1][j]
    if dp[i][j] >= modulo:
      dp[i][j] -= modulo

echo dp[n][8]
